File: app/src/main/java/com/nacoda/moviesmvvm/mvvm/detail/DetailViewModel.java
package com.nacoda.moviesmvvm.mvvm.detail;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.support.annotation.NonNull;

import com.nacoda.moviesmvvm.data.source.MoviesRepository;

/**
 * Created by Mayburger on 1/9/18.
 */

public class DetailViewModel extends AndroidViewModel {

    private MoviesRepository moviesRepository;

    public DetailViewModel(@NonNull Application application, MoviesRepository moviesRepository) {
        super(application);
        this.moviesRepository = moviesRepository;
    }
}

File: app/src/main/java/com/nacoda/moviesmvvm/mvvm/detail/DetailViewModel.kt
package com.nacoda.moviesmvvm.mvvm.detail

import android.annotation.SuppressLint
import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import com.nacoda.moviesmvvm.data.source.MoviesRepository

/**
 * Created by irfanirawansukirman on 04/12/17.
 */

class DetailViewModel(context: Application, private val mMoviesRepository: MoviesRepository) : AndroidViewModel(context) {

    @SuppressLint("StaticFieldLeak")
    var mContext = context
}


